pilot/pkg/bootstrap/server.go

```diff
@@ -678,10 +677,8 @@ func (s *Server) waitForCacheSync(stop <-chan struct{}) bool {
 	// TODO: remove dependency on k8s lib
 	// TODO: set a limit, panic otherwise (to not hide the error)
 	if !cache.WaitForCacheSync(stop, func() bool {
-		if s.kubeRegistry != nil {
-			if !s.kubeRegistry.HasSynced() {
```

Comment:
nit: remove the TODO above - I think it is for this only

Reply:
ok